### PR TITLE
user database snapshot for particular timestamp

### PR DESCRIPTION
**Pull request by prettyirrelevant**
This initial commit is a status report aimed at collecting feedback/review.

### CODE REVIEW

File: rotkehlchen/api/v1/schemas.py [hunk 1]
```diff
@@ -1951,3 +1951,7 @@ def validate_schema(  # pylint: disable=no-self-use
 
 class AssetsImportingFromFormSchema(Schema):
     file = FileField(allowed_extensions=['.zip', '.json'], required=True)
+
+
+class DBSnapshotQuerySchema(Schema):
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1956]: This is not wrong, but we can just simplify the code a bit.

There is already a schema with the exact same schema (just a single timestamp) here: https://github.com/rotki/rotki/blob/963c13219c308f1efd19724e6793a1013482ddc5/rotkehlchen/api/v1/schemas.py#L1764-L1765

How about renaming it from `TimedManualPriceSchema` to something like `SingleTimestampSchema` and then use that instead in the resources you made and wherever `TimedManualPriceSchema` was used?

File: rotkehlchen/db/dbhandler.py [hunk 2]
```diff
@@ -3494,3 +3494,39 @@ def get_associated_locations(self) -> Set[Location]:
         if cursor.fetchone()[0] >= 1:
             locations.add(Location.BALANCER)
         return locations
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[SingleDBAssetBalance]:
```
Comment (LefterisJP) [inline rotkehlchen/db/dbhandler.py:3498]: There is already code in dbhandler that queries timed balances here: https://github.com/rotki/rotki/blob/963c13219c308f1efd19724e6793a1013482ddc5/rotkehlchen/db/dbhandler.py#L2770

Why not use that?

File: rotkehlchen/api/rest.py [hunk 3]
```diff
@@ -4062,3 +4063,74 @@ def import_user_assets(self, path: Path) -> Response:
...
+        processed_timed_location_data = process_result_list(timed_location_data)
+
+        timed_balances = self.rotkehlchen.data.db.get_timed_balances(timestamp)
+        processed_time_balances_data = process_result_list(timed_balances)
+
+        tmp_dir = tempfile.TemporaryDirectory()
+        tmp_dir_path = Path(tmp_dir.name)
+        timed_balances_csv_path = tmp_dir_path / 'timed_balances.csv'
+        timed_location_data_csv_path = tmp_dir_path / 'timed_location_data.csv'
+        snapshot_zip_path = tmp_dir_path / 'snapshot.zip'
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:4079]: This is a lot of code to be here.

And it also duplicates in the `download_db_user_snapshot()` function below.

You know what would be a good idea? If you made a new DB class under https://github.com/rotki/rotki/tree/develop/rotkehlchen/db in a new file called snapshots.py. This way you could abstract all the code there, along with the changes you put in dbhandler.py  (we try to break stuff out of that huge file, not add more :smile: ).

This way you can abstract the entire fucntionality there and call it from here and from `download_user_snapshot` with different arguments denoting the different way of getting the CSV.

This will also be helpful since we have more work to be done with the snapshots in the future, such as deleting certain snapshots or editing them.

File: rotkehlchen/db/dbhandler.py [hunk 4]
```diff
@@ -3494,3 +3494,39 @@ def get_associated_locations(self) -> Set[Location]:
...
+        balances_data = []
+        cursor = self.conn.cursor()
+        timed_balances_result = cursor.execute(
+            'SELECT category, time, amount, usd_value FROM timed_balances '
+            'WHERE time=?', (timestamp,),
+        )
+        for data in timed_balances_result:
+            balances_data.append(
+                SingleDBAssetBalance(
```
Comment (LefterisJP) [inline rotkehlchen/db/dbhandler.py:3507]: Don't use SingleDBAssetBalance.

The structure to return should contain a list of balance type, asset, amount, and usd values. This one does not contain the asset.

You probably want `DBAssetBalance` but that still contains timestamp. You can probably just ignore the timestamp at export. But you should create a `serialize()` function to serialize it for CSV purposes. Don't use `process_result_list()` as that will result in the asset identifier being used which is not human readable. Instead it should be `str(asset)` for the asset, which will create a human readable represntation`.

**Review by LefterisJP — COMMENTED** [review]
Hey @prettyirrelevant thanks. Left some initial comments on the direction.

File: rotkehlchen/db/dbhandler.py (hunk 2 above)
Comment (prettyirrelevant) [reply rotkehlchen/db/dbhandler.py:3498]: The issue with using `query_timed_balances` is the fact that it defines an `asset` parameter. The constraint with that is having to add multiple lists together.

I might be wrong though.

**Comment by prettyirrelevant** [comment]
If my implementation so far is fine, can I go ahead with writing tests?
Noob question: In testing the endpoint, is checking the status code & mimetype sufficient(the case of downloading the snapshot)?

Also, in the case of exporting the snapshot, I read the code that exports processed history csv. It accepts a `directory_path` argument. How does that work please?

**Comment by LefterisJP** [comment]
> If my implementation so far is fine, can I go ahead with writing tests? Noob question: In testing the endpoint, is checking the status code & mimetype sufficient(the case of downloading the snapshot)?
> 
> Also, in the case of exporting the snapshot, I read the code that exports processed history csv. It accepts a `directory_path` argument. How does that work please?

Why are you adding v31_rotkehlchen.db in this PR? There is a conflict. Tests and implementation are separate. You can always write tests if you know what you are supposed to be expecting from what you build.

**Comment by prettyirrelevant** [comment]
> > If my implementation so far is fine, can I go ahead with writing tests? Noob question: In testing the endpoint, is checking the status code & mimetype sufficient(the case of downloading the snapshot)?
> > Also, in the case of exporting the snapshot, I read the code that exports processed history csv. It accepts a `directory_path` argument. How does that work please?
> 
> Why are you adding v31_rotkehlchen.db in this PR? There is a conflict. Tests and implementation are separate. You can always write tests if you know what you are supposed to be expecting from what you build.

My bad, apologies, I just do `git add .`

**Comment by LefterisJP** [comment]
> > > If my implementation so far is fine, can I go ahead with writing tests? Noob question: In testing the endpoint, is checking the status code & mimetype sufficient(the case of downloading the snapshot)?
> > > Also, in the case of exporting the snapshot, I read the code that exports processed history csv. It accepts a `directory_path` argument. How does that work please?
> > 
> > 
> > Why are you adding v31_rotkehlchen.db in this PR? There is a conflict. Tests and implementation are separate. You can always write tests if you know what you are supposed to be expecting from what you build.
> 
> My bad, apologies, I just do `git add .`

Don't do that. Add the files you need and use a git frontend or something to make it easier for you. Your editor should have something (hopefully)

File: rotkehlchen/db/snapshots.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    def clean_data(self, entries: List[Any]) -> List[Dict[str, Any]]:
+        new_entries = []
+        for entry in entries:
+            new_entries.append(self.serialize_data_for_csv(entry))
+
+        return new_entries
+
+    # pylint: disable=no-self-use
+    def serialize_data_for_csv(self, entry: Any) -> Dict[str, Any]:
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:93]: Since you have the no-self-use warning you can just make this into a static method.

```suggestion
    @staticmethod
    def serialize_data_for_csv(entry: Any) -> Dict[str, Any]:
```

File: rotkehlchen/db/snapshots.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:23]: Please be adding a docstring to most functions with a quick explanation of what it does (one line is fine) and then just as we do in other functions, mention the possible exceptions it may raise.

Mentioning exceptions in the docstring is a habit we have started doing which helps us understand what errors need to be handled where.

Do it for most functions in this file.

File: rotkehlchen/db/snapshots.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    def clean_data(self, entries: List[Any]) -> List[Dict[str, Any]]:
+        new_entries = []
+        for entry in entries:
+            new_entries.append(self.serialize_data_for_csv(entry))
+
+        return new_entries
+
+    # pylint: disable=no-self-use
+    def serialize_data_for_csv(self, entry: Any) -> Dict[str, Any]:
+        if isinstance(entry, LocationData):
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:94]: This function is not needed. What you should do instead is add a `serialize()` function to the `LocationData` `NamedTuple` and one to the `DBAssetBalance`

File: rotkehlchen/db/snapshots.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            balances_data.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    asset=data[3],
+                    usd_value=data[4],
+                ),
+            )
+        return self.clean_data(balances_data)
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:40]: Don't return cleaned data from here. Do it only at the api exit level. This should return a list of `DBAssetBalance()`

File: rotkehlchen/db/snapshots.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+            with ZipFile(self.snapshot_archive_path, mode='w') as archive:
+                archive.write(self.timed_balances_csv_path)
+                archive.write(self.timed_location_data_csv_path)
+
+            return True, str(self.snapshot_archive_path)
+        except (CSVWriteError, PermissionError) as e:
+            return False, str(e)
+
+    def clean_data(self, entries: List[Any]) -> List[Dict[str, Any]]:
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:85]: This function is not needed after the changes I suggested.

File: rotkehlchen/db/dbhandler.py [hunk 10]
```diff
@@ -3494,3 +3494,39 @@ def get_associated_locations(self) -> Set[Location]:
...
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    usd_value=data[3],
+                ),
+            )
+        return balances_data
+
+    def get_timed_location_data(self, timestamp: Timestamp) -> List[LocationData]:
```
Comment (LefterisJP) [inline rotkehlchen/db/dbhandler.py:3516]: Why is this function still here? You moved it to db/snapshot.py.

File: rotkehlchen/db/dbhandler.py (hunk 2 above)
Comment (LefterisJP) [inline rotkehlchen/db/dbhandler.py:3498]: Why is this function still here? You moved it to db/snapshot.py.

File: rotkehlchen/api/v1/schemas.py [hunk 11]
```diff
@@ -1743,8 +1743,11 @@ class ManualPriceSchema(Schema):
     price = PriceField(required=True)
 
 
-class TimedManualPriceSchema(ManualPriceSchema):
+class SingleTimestampSchema(ManualPriceSchema):
     timestamp = TimestampField(required=True)
+    from_asset = AssetField(required=False)
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1748]: What are you doing here? This is not what I told you to do: https://github.com/rotki/rotki/pull/4187#discussion_r850336293

I said to just rename it to `SingleTimestampSchema`, not to add 3 more fields. Why did you add those?

File: rotkehlchen/db/snapshots.py [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+        for data in timed_location_data:
+            location_data.append(
+                LocationData(
+                    time=data[0],
+                    location=data[1],
+                    usd_value=data[2],
+                ),
+            )
+
+        return self.clean_data(location_data)
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:59]: same here. Make this return a list of `LocationData`. Do not serialize the data here already. Do it at api level.

File: rotkehlchen/db/snapshots.py [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:21]:
```suggestion
```

Remove those. They should not be initialized here. This is a temporal object. The temporary directory and files should only be created when/if needed, not at initiailzation.

File: rotkehlchen/db/snapshots.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+                ),
+            )
+
+        return self.clean_data(location_data)
+
+    def export(
+        self,
+        timed_balances_data: List[Dict[str, Any]],
+        timed_location_data: List[Dict[str, Any]],
+    ) -> Tuple[bool, str]:
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:65]: So this function is very wrong. You should have copied what we do with the accountant export or at least asked me. Look at what we do already and keep things consistent with the codebase. Please do:

1. Pass the timestamp to export
2. Pass an optional `directory_path: Optional[Path]`. If given then the CSV is exported to that directory. If not given a temporary directory is created.
3. Remove `timed_balances_data` and `timed_location_data` as arguments. These can be calculated inside the function itself.
4. Move the creation of the directory or usage of passed directory here.
5. Add a docstring explaining what this does and what it returns

File: rotkehlchen/rotkehlchen.py [hunk 15]
```diff
@@ -309,6 +310,7 @@ def unlock_user(
             evm_tx_decoder=self.evm_tx_decoder,
             premium=self.premium,
         )
+        self.snapshot = Snapshot(self.data.db)
```
Comment (LefterisJP) [inline rotkehlchen/rotkehlchen.py:313]: Don't initialize this here. This should be a temporal object.

File: rotkehlchen/api/rest.py [hunk 16]
```diff
@@ -4062,3 +4062,38 @@ def import_user_assets(self, path: Path) -> Response:
             )
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    @require_loggedin_user()
+    def export_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        timed_location_data = self.rotkehlchen.snapshot.get_timed_location_data(timestamp)
+        timed_balances = self.rotkehlchen.snapshot.get_timed_balances(timestamp)
+
+        is_success, message = self.rotkehlchen.snapshot.export(timed_balances, timed_location_data)
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:4071]: If you do all I told you this should be replaced by:

```suggestion
    def export_user_db_snapshot(self, timestamp: Timestamp, path: Path) -> Response:
        dbsnapshots = DBSnapshot(self.rotkehlchen.data.db)
        is_success, message = snapshot.export(timestamp=timestamp, path=path)
```

File: rotkehlchen/api/rest.py [hunk 17]
```diff
@@ -4062,3 +4062,38 @@ def import_user_assets(self, path: Path) -> Response:
...
+        return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    @require_loggedin_user()
+    def download_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        timed_location_data = self.rotkehlchen.snapshot.get_timed_location_data(timestamp)
+        timed_balances = self.rotkehlchen.snapshot.get_timed_balances(timestamp)
+
+        is_success, zipfile_path = self.rotkehlchen.snapshot.export(
+            timed_balances, timed_location_data,
+        )
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:4084]: If you do what I told you this should be replaced by

```suggestion
       dbsnapshots = DBSnapshot(self.rotkehlchen.data.db)
       is_success, zipfile_path = snapshot.export(timestamp=timestamp, path=None)
```

**Review by LefterisJP — COMMENTED** [review]
A lot of things to fix. Not close to what I meant unfortunately.

File: docs/api.rst [hunk 18]
```diff
@@ -10312,3 +10312,71 @@ Import assets added by the user
...
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /api/1/snapshot/export?timestamp=133899009&path=/home/user/Documents HTTP/1.1
+      Host: localhost:5042
+
+   :param str path: The directory in which to write the exported CSV files
+   :param str timestamp: The epoch timestamp representing the day of the snapshot to be returned
```
Comment (LefterisJP) [inline docs/api.rst:10333]:
```suggestion
      POST /api/1/snapshot/export HTTP/1.1
      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {
          "path": "/home/user/Documents",
          "timestamp": 133899009
      }

   :reqjson str path: The directory in which to write the exported CSV files
   :reqjson int timestamp: The epoch timestamp representing the time of the snapshot to be returned
```

Also note the type of the timestamp parameter is a integer

File: docs/api.rst [hunk 19]
```diff
@@ -10312,3 +10312,71 @@ Import assets added by the user
...
+   Doing a GET on the snapshot download endpoint will download database snapshot for the specified timestamp as a zip file.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /api/1/snapshot/export?timestamp=133899009 HTTP/1.1
+      Host: localhost:5042
+
+   :param str timestamp: The epoch timestamp representing the day of the snapshot to be returned
```
Comment (LefterisJP) [inline docs/api.rst:10369]: Please also change this to POST as above

File: docs/api.rst [hunk 20]
```diff
@@ -10312,3 +10312,71 @@ Import assets added by the user
...
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/zip
+
+
+   :statuscode 200: Snapshot was downloaded successfully.
+   :statuscode 400: Provided timestamp query paramater is missing.
```
Comment (LefterisJP) [inline docs/api.rst:10380]:
```suggestion
   :statuscode 400: Provided timestamp query parameter is missing.
```

File: rotkehlchen/api/v1/schemas.py (hunk 11 above)
Comment (LefterisJP) [reply rotkehlchen/api/v1/schemas.py:1766]: ping. You have not answered this comment. What are you doing here?

File: rotkehlchen/db/dbhandler.py (hunk 2 above)
Comment (LefterisJP) [reply rotkehlchen/db/dbhandler.py:3498]: ping ...?

File: rotkehlchen/db/dbhandler.py (hunk 10 above)
Comment (LefterisJP) [reply rotkehlchen/db/dbhandler.py:3516]: ping .... ?

File: rotkehlchen/db/snapshots.py [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+        success = False
+        filename = ''
+        if archive.filename is not None:
+            success = True
+            filename = archive.filename
+
+        return success, filename
+
+    def export(self, timestamp: Timestamp, directory_path: Optional[Path]) -> Tuple[bool, str]:
+        """Export the database's snapshot for specified timestamp."""
```
Comment (LefterisJP) [inline rotkehlchen/db/snapshots.py:96]: Add a bit more info in the docstring here, explaining the difference in behavior between providing the directory_path and not.

File: rotkehlchen/tests/api/test_snapshots.py [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+
+
+def assert_csv_export_response(response, csv_dir, is_download=False):
+    if is_download:
+        assert response.status_code == HTTPStatus.OK
+    else:
+        assert_proper_response(response)
+        data = response.json()
+        assert data['message'] == ''
+        assert data['result'] is True
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:63]: We have a utility test function in tests/utils/api.py that does that all.

```suggestion
        assert_simple_ok_response(response)
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+                'nexo',
+                'blockfi',
+                'independentreserve',
+                'gitcoin',
+                'sushiswap',
+                'uphold',
+                'bitpanda',
+                'bisq',
+                'ftxus',
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:120]: You can check that in a much easier way
```suggestion
            assert Location.deserialize(row['location']) is not None
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 24]
```diff
@@ -0,0 +1,178 @@
...
+            assert row['usd_value'] is not None
+            count += 1
+        assert count == 3
+
+
+def test_export_snapshot(rotkehlchen_api_server, user_data_dir, tmpdir_factory):
+    msg_aggregator = MessagesAggregator()
+    db = DBHandler(user_data_dir, '123', msg_aggregator, None)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:128]: No no don't create another DB here. The `rotkehlchen_api_server` is a full rotki instance and has a db embedded in it.

```suggestion
    db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
```

File: rotkehlchen/api/v1/resources.py [hunk 25]
```diff
@@ -2081,3 +2082,19 @@ def post(self, file: FileStorage) -> Response:
             file.save(str(filepath))
             response = self.rest_api.import_user_assets(path=filepath)
         return response
+
+
+class DBSnapshotExportingResource(BaseResource):
+    get_schema = SingleTimestampExportingSchema()
+
+    @use_kwargs(get_schema, location='query')
+    def get(self, timestamp: Timestamp, path: Path) -> Response:
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/resources.py:2091]: As I mentioned in the docs comments, please change both of these to post

File: rotkehlchen/tests/api/test_snapshots.py [hunk 26]
```diff
@@ -0,0 +1,178 @@
...
+    csv_dir2 = str(tmpdir_factory.mktemp('test_csv_dir2'))
+    _populate_db_with_balances(conn)
+    _populate_db_with_location_data(conn)
+
+    # test with query params
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'dbsnapshotexportingresource',
+            timestamp='1',
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:141]: timestamp should be an integer here and below

File: rotkehlchen/tests/api/test_snapshots.py [hunk 27]
```diff
@@ -0,0 +1,178 @@
...
+            path=csv_dir2,
+        ),
+    )
+    assert_csv_export_response(response, csv_dir2, is_download=False)
+
+
+def test_download_snapshot(rotkehlchen_api_server, user_data_dir):
+    msg_aggregator = MessagesAggregator()
+    db = DBHandler(user_data_dir, '123', msg_aggregator, None)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:159]: same comment I made in the previous test for the DB.

**Review by LefterisJP — COMMENTED** [review]
More stuff needed, but it looks bit better now. There is still quite a few comments from my previous PR review though about code that should be removed that have not been done. Please when you ask me to re-review, make sure you have addressed all comments.

@prettyirrelevant

File: rotkehlchen/tests/api/test_snapshots.py [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+            rotkehlchen_api_server,
+            'dbsnapshotexportingresource',
+        ),
+        json={
+            'timestamp': 1,
+            'path': csv_dir,
+        },
+    )
+    assert_csv_export_response(response, csv_dir, is_download=False)
+    # test query params
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:105]: I would remove the query params both as a test and as an option. I realize we have it in other non-get endpoints, though so we can do it in a different PR.

File: rotkehlchen/api/v1/resources.py [hunk 29]
```diff
@@ -2081,3 +2083,19 @@ def post(self, file: FileStorage) -> Response:
             file.save(str(filepath))
             response = self.rest_api.import_user_assets(path=filepath)
         return response
+
+
+class DBSnapshotExportingResource(BaseResource):
+    get_schema = SnapshotExportingSchema()
+
+    @use_kwargs(get_schema, location='json_and_query')
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/resources.py:2091]: I would remove the query params. I realize we have it in other non-get endpoints, though so we can do it in a different PR.

**Review by LefterisJP — APPROVED** [review]
Hey @prettyirrelevant so I left one more comment on not using query params with a POST endpoint as they are not needed and our frontend is not using them (I think).

But this is minor and we can adjust in another PR.

Now for the failing tests, they pass locally and it seems that all eth2/beaconchain tests failed due to cloudflare errors and 2 tests had etherscan hiccups. They are unrelated to your PR and they pass locally.

I am going to merge.
